### PR TITLE
Adding a validation to check if base resources are already deployed.

### PR DESCRIPTION
**Pull request by rkaraddi**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
----------
#### User story for this change
https://dev.azure.com/msazuredev/AzureForOperators/_workitems/edit/1161413
------------
#### Ran the tests locally:
image
#### Ran the style test locally:
image
#### Ran the linter test locally:
image
-----------
##### When NFD group(or any base resource) is not present the base resources are redeployed
nfdv publish without nfdg
2 nfd deployments

##### When all base resources are present then only NFDV get deployed
only nfdv deploy

##### When NSD Group(or any base resource) is not present then all base resources are deployed
nsd old

##### When all base resources are present then only CGS and NSDV get deployed
nsd fast

##### There is an improvement of ~ 4 mins in command runtime

--------

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py [hunk 1]
```diff
@@ -153,6 +159,72 @@ def _artifact_manifests_exist(
 
         return ManifestsExist.NONE
 
+    @staticmethod
+    def _base_Resources_exist(
```
Comment (jordlay) [inline src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py:163]: Could you please make the R lowercase to match the style of the rest

File: src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py [hunk 2]
```diff
@@ -153,6 +159,72 @@ def _artifact_manifests_exist(
 
         return ManifestsExist.NONE
 
+    @staticmethod
+    def _base_Resources_exist(
+        config: BaseCommonParametersConfig, command_context: CommandContext
+    ) -> bool:
+        """
+
+        Returns True if all required manifests exist, False if none do, and raises an
```
Comment (jordlay) [inline src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py:168]: Think this is a typo, from copying over the docstring from the check manifest exists, should be return true if all base resourcs exist

**Comment by jordlay** [comment]
Logic looks good. I think its a code smell that we are have so many if statement for instance checking, however, that's not no you to fix. We also do it in the artifact manifest exist code. You may need to run your azdev commands again now that I've asked you to change the target branch to b2 release. Could you also add to the PR if you have done any manual testing? Like successfully publishing an nfdv without skipping base and with? Looks good!

File: src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py (hunk 2 above)
Comment (rkaraddi) [reply src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py]: Yeah, forgot to change it. I have fixed it now.

File: src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py (hunk 1 above)
Comment (rkaraddi) [reply src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py]: changed this.

**Comment by jordlay** [comment]
You should also update the history.RST under the b2 release with your change

File: src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py [hunk 3]
```diff
@@ -153,6 +159,71 @@ def _artifact_manifests_exist(
 
         return ManifestsExist.NONE
 
+    @staticmethod
```
Comment (jordlay) [inline src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py:162]: Now that you have implemented the skipping base, can you remove the big comments from lines 232 to 237

**Comment by jordlay** [comment]
Rakshita and I have chatted, we need to make a few changes before this is ready. Will rereview after that :)

File: src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py (hunk 3 above)
Comment (rkaraddi) [reply src/aosm/azext_aosm/definition_folder/reader/bicep_definition.py:162]: removed those lines

**Pull request merged**